students/ptfbanks/Lesson03/pf_mailroom.py

```diff
@@ -0,0 +1,93 @@
+#  MailRoom exercise lesson03 final task
+#    - Build, sort, print a table
+#--------------------------Global Region-----------------------------------#
+donations = [
+    ("Bob Williams", 5580.00),
+    ("Tom Haskell", 345.05),
+    ("Earl Jackson", 4553.45),
+    ("Les Thomas", 8934.00),
+    ("James Black", 154),
+    ("Bob Williams", 3245.50),
+    ("Tom Haskell", 3245.90),
+    ("Earl Jackson", 45.6),
+    ("Les Thomas", 325.00),
+    ("James Black", 1300),
+    ("Bob Williams", 1000),
+    ("Tom Haskell", 243.05),
+    ("Earl Jackson", 345.00),
+    ("Les Thomas", 3665.00),
+    ("James Black", 21)]
```

Comment:
in real applications choosing the right data structure is critical - and as you are learning it may be a trial and error.
it's important to know all available basic data structures and then use one or combine them.

for example a list can contain pretty much anything and get really nested:
`some_list = [1, (1, 2, 4), {"key1": "value1"}, Donor(), ['some-strings']]`

that one contains: integer, tuple, dictionary, object and a list - there is really no limit.

For the mailroom program you get to decide what makes the most sense, keeping in mind that you do want to keep your code simple and not over-complicate it. 

I would suggest following data structure where donations are flat and you have 1 unique donor in the list:
```
donors = [("james smith", [100, 200, 300]), ("les thomas", [343, 33])]
```
it should really simplify your code.